presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -419,15 +421,15 @@ else if (parameterType.isPrimitive()) {
 
                     boolean currentNullFlag = false;
                     if (i + 1 < method.getParameterCount()) {
-                        boolean foundIsNull = false;
                         annotations = method.getParameterAnnotations()[i + 1];
-                        for (Annotation annotation : annotations) {
-                            if (annotation instanceof IsNull) {
-                                foundIsNull = true;
-                                break;
+                        if (Stream.of(annotations).anyMatch(IsNull.class::isInstance)) {
+                            Class<?> isNullType = method.getParameterTypes()[i + 1];
+
+                            checkArgument(isNullType == boolean.class, "Method [%s] has parameter with @IsNull that is type %s", method, isNullType.getSimpleName());
```

Comment:
We can simplify the message and also give the user a better indication what is wrong: 
```java
"Method [%s] has non-boolean parameter with @IsNull"